cloudify_hostpool/storage/base.py

```diff
@@ -58,20 +58,20 @@ def update_server(self, global_id, server):
         :param global_id: The global id of the server.
         :param server: The server dictionary containing the new values.
 
-        :return False if no change occurred, True otherwise.
-        :rtype bool
+        :return  None if no change occurred,, server otherwise.
+        :rtype  dict
 
         """
         pass
 
     @abc.abstractmethod
-    def reserve_server(self, **filters):
+    def reserve_server(self, global_id):
```

Comment:
now something doesn't add up.
i though reserve server accepts filter options and reserves the first server that meets those filters.
notice the method documentation now does not fit..

what exactly should it do? and what do you think about my proposal to pass a 'template' parameters to functions that we want to have filtering ability?